spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -279,10 +291,14 @@ private void setUpSSL() throws Exception {
 			TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
 			tmf.init(tks);
 
-			SSLContext context = SSLContext.getInstance("SSLv3");
+			SSLContext context = createSSLContext();
 			context.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
 			this.connectionFactory.useSslProtocol(context);
 		}
 	}
 
+	protected SSLContext createSSLContext() throws NoSuchAlgorithmException {
```

Comment:
Nothing to complain about.
Not sure that we must document it, but the good JavaDoc here, as well as on the class level, would be enough for end-users.